services/src/pro/layers/postgres_layer_db.rs

```diff
@@ -331,6 +334,123 @@ where
 
         Ok(())
     }
+
+    async fn remove_collection(&self, collection: &LayerCollectionId) -> Result<()> {
+        let collection =
+            Uuid::from_str(&collection.0).map_err(|_| crate::error::Error::IdStringMustBeUuid {
+                found: collection.0.clone(),
+            })?;
+
+        if collection == INTERNAL_LAYER_DB_ROOT_COLLECTION_ID {
+            return Err(LayerDbError::CannotRemoveRootCollection.into());
+        }
+
+        let conn = self.conn_pool.get().await?;
+
+        // delete the collection!
+        // on delete cascade removes all entries from `collection_children` and `collection_layers`
+
+        let remove_layer_collection_stmt = conn
+            .prepare(
+                "DELETE FROM layer_collections
+                 WHERE id = $1;",
+            )
+            .await?;
+        conn.execute(&remove_layer_collection_stmt, &[&collection])
+            .await?;
+
+        // delete all collections without parent collection
+        // TODO: use recursive delete statement
```

Comment:
I thought we determined this to be infeasible?

Reply:
[c48b6ec](https://github.com/geo-engine/geoengine/pull/676/commits/c48b6ec12e37040016e367af4fbb230bf3b6923f)